### DISCUSSION TITLE
Almost no context

### DISCUSSION THREAD

**Discussion by j-taptok** [open-webui/open-webui#1345] [Q&A]
What could be reason that the AI loses context very quick?

We are talking about specific topic, then I ask something without proving context about the previous topic and it replies with something new not related to the topic we were discussing or it doesn't continue the conversation. I have to keep copy and pasting past conversations for it to follow it.

Very frustrating, does anyone knows if this is a parameter that I am missing?

**Comment by ghost** [comment]
This depends on which model and back-end system you're using. Are you using Ollama locally or something else remotely? There are some advanced parameters you can try under _Settings -> General -> Advanced Parameters_ like **Context Length**:
![image](https://github.com/open-webui/open-webui/assets/126992880/7a0b4fe4-d833-4e88-9c4d-e4b084742a9a)

**Reply by gappie** [reply]
What is the default context length when using ollama locally? If I change the context length under the advanced settings, does that change the value for all loaded models?

**Reply by ghost** [reply]
Depends on which model you are using. Default context in OWUI is only 128 tokens. Go to the menu as shown above and increase that to at least 2048. You can go to 8192 for Llama3 untuned. That's possibly your issue. The prompt is being limited to a very small number of tokens in OWUI default.

**Reply by justinh-rahb** [reply]
> Depends on which model you are using. Default context in OWUI is only 128 tokens. Go to the menu as shown above and increase that to at least 2048. You can go to 8192 for Llama3 untuned. That's possibly your issue. The prompt is being limited to a very small number of tokens in OWUI default.

Not sure where you pulled the number `128` from, but there's no such default value set by Open WebUI on the `num_ctx` parameter. If the Modelfile sets a value for this, that will be used, if the user _clicks_ the `Default` button shown in the above screenshot the frontend sets it to `2048` as shown:
https://github.com/open-webui/open-webui/blob/b3da09f52c3a80967bec5f6f5c638470cc20af8d/src/lib/components/chat/Settings/Advanced/AdvancedParams.svelte#L477-L479
https://github.com/search?q=repo%3Aopen-webui%2Fopen-webui%20num_ctx&type=code

If the value is set from **Advanced Parameters** it will override the Modelfile value if there is one. If it's kept on `Default` there is no value set.

**Comment by j-taptok** [comment]
@justinh-rahb Sonnet allows +100k context window, now I have 16k and it gives me this error:

<img width="686" alt="image" src="https://github.com/open-webui/open-webui/assets/98992496/a2a2b342-9cbd-4ba8-b323-2ea5b7ceb715">


<img width="482" alt="image" src="https://github.com/open-webui/open-webui/assets/98992496/5a16e05f-4dba-4c9e-99ef-3a6de94520a3">

If I leave it in "default" it works just fine but then it forgets if the conversation extends.

I tried with Azure, and similar error:

AzureException - Error code: 400 - {'error': {'message': 'Unrecognized request argument supplied: num_ctx. Please contact us through an Azure support request at: https://go.microsoft.com/fwlink/?linkid=2213926 for further questions.', 'type': 'invalid_request_error', 'param': None, 'code': None}}

<img width="785" alt="image" src="https://github.com/open-webui/open-webui/assets/98992496/067e3ae3-14bd-4f60-97cf-e3917d76dcb0">

**Reply by justinh-rahb** [reply]
@j-taptok `num_ctx` parameter is generally not supported by any external APIs, the context of models on OpenAI, Azure, Anthropic and others are set by them. However, we can't just drop this param from LiteLLM calls like we do for OpenAI connections as LiteLLM does support some providers that _do_ use this parameter.

Now I see where you got `128` from though, as that is the default `max_tokens` returned parameter for **Anthropic API** in particular if you do not specify otherwise, please see this note:
![Screenshot 2024-04-24 at 12 15 28 PM](https://github.com/open-webui/open-webui/assets/52832301/c64fb521-80e3-4a69-b2fb-7e6c3ddd10cb)
- #1038

**Reply by j-taptok** [reply]
I have 3000 in max_tokens, but the problem is context windows, something is off because it forgets the conversation after 4-5 prompts. 

<img width="494" alt="image" src="https://github.com/open-webui/open-webui/assets/98992496/4235fb21-26ae-478b-88cd-992743bb53b3">

I believe it happens mostly when I attach a #FileName , it starts having it as context but after 2-4 prompts it forgets it. Does that makes sense?

**Reply by justinh-rahb** [reply]
> I believe it happens mostly when I attach a #FileName , it starts having it as context but after 2-4 prompts it forgets it. Does that makes sense?

This has been observed, it's best to keep RAG conversations short in my experience, or they fall apart, doesn't seem to matter what you use (local, external API)

**Comment by j-taptok** [comment]
> > I believe it happens mostly when I attach a #FileName , it starts having it as context but after 2-4 prompts it forgets it. Does that makes sense?
> 
> This has been observed, it's best to keep RAG conversations short in my experience, or they fall apart, doesn't seem to matter what you use (local, external API)

I understand, is this a limitation of open-webui or just the way RAG is setup? Perhaps for now I can just paste the content of the RAG directly as text and work out the context that way.
